### ISSUE TITLE
add ability to append data to all trials

### ISSUE THREAD

**Issue by jodeleeuw** [jodeleeuw/jsPsych#136]
Often it's necessary to append some kind of data to all of the trials. For example, a subject number or condition assignment. This should be easier. Perhaps:

```javascript
jsPsych.data.addProperty({
  subjectid: 2,
  condition: 'left-handed'
});
```

List can be arbitrary length with arbitrary key: value pairs. Will update all of the data currently stored, plus add it to any new data that comes in (allowing the method to be called before or after experiment.

**Comment by jodeleeuw** [comment]
Related to this: remove the append_data parameter from dataAsCSV.

**Comment by jodeleeuw** [comment]
Implemented on /dev. Will be released in v4.2
